File: src/main/java/destructuringdeclaration/java/DestructuringDeclaration.java
package destructuringdeclaration.java;

import dataclass.java.Search;
import java.util.concurrent.atomic.AtomicInteger;

@SuppressWarnings("ALL")
public final class DestructuringDeclaration {
    public static final void main() {
        Search search = new Search("Kotlin", new AtomicInteger(100));
        String query = search.component1();
        AtomicInteger hits = search.component2();
        String string = "Found " + hits + " hits for query '" + query + '\'';
        System.out.println(string);
    }

    public static /* synthetic */ void main(String[] args) {
        main();
    }
}

File: src/main/kotlin/destructuringdeclaration/kotlin/DestructuringDeclaration.kt
package destructuringdeclaration.kotlin

import dataclass.kotlin.Search
import java.util.concurrent.atomic.AtomicInteger

fun main() {
    val (query: String, hits: AtomicInteger) = Search("Kotlin", AtomicInteger(100))
    println("Found $hits hits for query '$query'")
}
